client/state/themes/selectors.js

```diff
@@ -12,6 +13,235 @@ import config from 'config';
+ * @param  {Number}  siteId Site ID
+ * @param  {String}  themeId Theme ID
+ * @return {?Object}        Theme object
+ */
+export const getTheme = createSelector(
+	( state, siteId, themeId ) => {
+		const manager = state.themes.queries[ siteId ];
+		if ( ! manager ) {
+			return null;
+		}
+
+		return manager.getItem( themeId );
+	},
+	( state ) => state.themes.queries
+);
+
+/**
+ * Returns an array of normalized themes for the themes query, or null if no
+ * themes have been received.
+ *
+ * @param  {Object}  state  Global state tree
+ * @param  {Number}  siteId Site ID
+ * @param  {Object}  query  Theme query object
+ * @return {?Array}         Themes for the theme query
+ */
+export const getThemesForQuery = createSelector(
+	( state, siteId, query ) => {
+		const manager = state.themes.queries[ siteId ];
+		if ( ! manager ) {
+			return null;
+		}
+
+		const themes = manager.getItems( query );
+		if ( ! themes ) {
+			return null;
+		}
+
+		// ThemeQueryManager will return an array including undefined entries if
+		// it knows that a page of results exists for the query (via a previous
+		// request's `found` value) but the items haven't been received. While
+		// we could impose this on the developer to accommodate, instead we
+		// simply return null when any `undefined` entries exist in the set.
+		if ( includes( themes, undefined ) ) {
+			return null;
+		}
+
+		return themes;
+	},
+	( state ) => state.themes.queries,
+	( state, siteId, query ) => getSerializedThemesQuery( query, siteId )
+);
+
+/**
+ * Returns true if currently requesting themes for the themes query, or false
+ * otherwise.
+ *
+ * @param  {Object}  state  Global state tree
+ * @param  {Number}  siteId Site ID
+ * @param  {Object}  query  Theme query object
+ * @return {Boolean}        Whether themes are being requested
+ */
+export function isRequestingThemesForQuery( state, siteId, query ) {
+	const serializedQuery = getSerializedThemesQuery( query, siteId );
+	return !! state.themes.queryRequests[ serializedQuery ];
+}
+
+/**
+ * Returns the total number of items reported to be found for the given query,
+ * or null if the total number of queryable themes if unknown.
+ *
+ * @param  {Object}  state  Global state tree
+ * @param  {Number}  siteId Site ID
+ * @param  {Object}  query  Theme query object
+ * @return {?Number}        Total number of found items
+ */
+export function getThemesFoundForQuery( state, siteId, query ) {
+	if ( ! state.themes.queries[ siteId ] ) {
+		return null;
+	}
+
+	return state.themes.queries[ siteId ].getFound( query );
+}
+
+/**
+ * Returns the last queryable page of themes for the given query, or null if the
+ * total number of queryable themes if unknown.
+ *
+ * @param  {Object}  state  Global state tree
+ * @param  {Number}  siteId Site ID
+ * @param  {Object}  query  Theme query object
+ * @return {?Number}        Last themes page
+ */
+export function getThemesLastPageForQuery( state, siteId, query ) {
```

Comment:
Probably not. However, given their more digestible sizes, I think that it makes sense to keep them broken into  separate selectors that we also cover with tests separately.